src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -445,15 +445,13 @@ private PackageRestoreInputs DetermineRestoreInputs()
 
             // Run inputs through msbuild to determine the 
             // correct type and find dependencies as needed.
-            DetermineInputsFromMSBuild(packageRestoreInputs);
-
-            return packageRestoreInputs;
+            return await DetermineInputsFromMSBuildAsync(packageRestoreInputs);
```

Comment:
Why modify `DetermineInputsFromMSBuildAsync(...)` to return `Task<PackageRestoreInputs>` and not simply `Task`?  There is an implication that the return value is different than the argument, but it isn't.